### PR TITLE
feat(datahub cli): DataHub CLI Quickstart 

### PR DESCRIPTION
**Pull request by jjoyce0510**
This PR introduces a way to quickstart DataHub using the command-line tool. This means that you no longer are required to clone the repository to quickly spin up an instance of DataHub and start ingesting DataHub.

The new quickstart process:

1. Install Docker
2. Install DataHub CLI tool 
3. Run `datahub docker quickstart` to deploy DataHub at localhost:9002 
4. Run `datahub docker ingest-sample-data` to ingest the bootstrap_mce.json sample data. 

Once DataHub is deployed, you'll see the following success message:

```
✔ DataHub is now running
Ingest some demo data using `datahub docker ingest-sample-data`,
or head to http://localhost:9002 (username: datahub, password: datahub) to play around with the frontend.
```

To clean all state, just run `datahub docker nuke`. 

Using the new quickstart, the minimum required docker containers will be deployed. (consumers, gms, frontend, neo, elastic, mysql) 

**Changes**
1. Adding `datahub docker quickstart`, `datahub docker ingest-sample-data`, & `datahub docker nuke` to Python CLI
2. Adding scripts to auto-generate a docker compose file used for quickstart, which includes only the minimum required containers and inlines all default env variables. 
3. Adding auto-generated, minimalistic docker-compose.quickstart.yml file which is fetched by the CLI on demand. 
4. Updates to documentation to reflect changes to quickstart process

**Acknowledgements** 
Huge thanks to @hsheth2 & @dexter-mh-lee for contributing to this effort. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -65,14 +65,21 @@ jobs:
       - name: Gradle build
         run: ./gradlew build -x check -x docs-website:build
       - name: Smoke test
-        run: |
```
Comment (jjoyce0510) [inline .github/workflows/build-and-test.yml:68]: @hsheth2 was this intentional?

File: smoke-test/smoke.sh [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 
-set -euxo pipefail
-
 python3 -m venv venv
 source venv/bin/activate
 pip install --upgrade pip wheel setuptools
 pip install -r requirements.txt
 
-(cd ../metadata-ingestion && ./scripts/codegen.sh)
+datahub docker quickstart \
```
Comment (jjoyce0510) [inline smoke-test/smoke.sh:19]: @hsheth2 Just noticed this. Amazing!

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (hsheth2) [reply .github/workflows/build-and-test.yml:68]: Yep - smoke.sh now runs the up command within the script

**Review by shirshanka — APPROVED** [review]
Well this is very exciting! Thanks @hsheth2 @jjoyce0510 !

**Pull request merged**